modules/db.ts

```diff
@@ -1,42 +1,53 @@
-import Dexie from 'dexie';
+import Dexie, { DBCore, DBCoreTable, DBCoreMutateRequest, Table } from 'dexie';
 
-const db = new Dexie('pantry-pal');
+export interface Item {
+  id: number;
+  name: string;
+  price: number;
+  isPurchased: boolean;
+  section?: string;
+  quantityUnit?: string;
+  createdAt?: string;
+  updatedAt?: string;
+}
+// class created to inform TS about our DB structure
+class PantryPalDatabase extends Dexie {
+  items!: Table<Item, number>;
 
-// define the structure of the db
-// if we change this after it's been created we need to increment the version and define it
-db.version(1).stores({
-  items: '++id, name, price, isPurchased',
-});
+  constructor() {
+    super('pantry-pal');
+    this.version(1).stores({
+      items: '++id, name, price, isPurchased',
+    });
 
-// next version
-// add: store section - should match an enum
-// enum values - produce, dairy, ethnic foods, meat dept, deli, baking, spice, snack, frozen, drinks
-// quantity unit - also an enum
-// enum - pounds, ounces, units, box
-// createdAt - date
-// updatedAt - date
-db.version(2).stores({
-  items:
-    '++id, name, price, isPurchased, section, quantityUnit, createdAt, updatedAt',
-});
+    this.version(2).stores({
+      items:
+        '++id, name, price, isPurchased, section, quantityUnit, createdAt, updatedAt',
+    });
+  }
+}
```

Comment:
Created a handy little custom class to extend `dexie` (indexdb) and inform TS about the structure of our tables, which right now is just the `items` table.